File: svws-webclient/core/src/asd/validate/schueler/ValidatorSsgSchuelerStammdatenGeschlecht.ts
import { ValidatorSsg00SchuelerStammdatenGeschlecht } from '../../../asd/validate/schueler/ValidatorSsg00SchuelerStammdatenGeschlecht';
import type { Supplier } from '../../../java/util/function/Supplier';
import { Class } from '../../../java/lang/Class';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorSsgSchuelerStammdatenGeschlecht extends Validator {


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param idGeschlecht     das Geschlecht des Schuelers
	 * @param kontext          der Kontext des Validators
	 */
	public constructor(idGeschlecht: Supplier<number | null>, kontext: ValidatorKontext) {
		super(kontext);
		this._validatoren.add(new ValidatorSsg00SchuelerStammdatenGeschlecht(idGeschlecht, kontext));
	}

	protected pruefe(): boolean {
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.schueler.ValidatorSsgSchuelerStammdatenGeschlecht';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.schueler.ValidatorSsgSchuelerStammdatenGeschlecht', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorSsgSchuelerStammdatenGeschlecht>('de.svws_nrw.asd.validate.schueler.ValidatorSsgSchuelerStammdatenGeschlecht');

}

export function cast_de_svws_nrw_asd_validate_schueler_ValidatorSsgSchuelerStammdatenGeschlecht(obj: unknown): ValidatorSsgSchuelerStammdatenGeschlecht {
	return obj as ValidatorSsgSchuelerStammdatenGeschlecht;
}

File: svws-asd/src/main/java/de/svws_nrw/asd/validate/schueler/ValidatorSsgSchuelerStammdatenGeschlecht.java
package de.svws_nrw.asd.validate.schueler;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import de.svws_nrw.transpiler.annotations.AllowNull;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung auf das Geschlecht bei den Stammdaten
 * eines Schuelers einer Schule aus.
 */
public final class ValidatorSsgSchuelerStammdatenGeschlecht extends Validator {

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param idGeschlecht     das Geschlecht des Schuelers
	 * @param kontext          der Kontext des Validators
	 */
	public ValidatorSsgSchuelerStammdatenGeschlecht(final @NotNull Supplier<@AllowNull Integer> idGeschlecht,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		_validatoren.add(new ValidatorSsg00SchuelerStammdatenGeschlecht(idGeschlecht, kontext));
	}

	@Override
	protected boolean pruefe() {
		return true;
	}

}
